src/encoded/static/components/dataset.js

```diff
@@ -3164,13 +3173,119 @@ DiseaseSeries.contextTypes = {
 globals.contentViews.register(DiseaseSeries, 'DiseaseSeries');
 
 
+/**
+ * The given array of cloning or mappings datasets get mapped to an array of each of these datasets
+ * along with their assemblies and default files. Datasets with analyses restrict the files to
+ * those they contain.
+ * @param {array} datasets Cloning or mappings datasets to map to their files and assemblies
+ * @param {array} allFiles File objects from all cloning and mappings datasets
+ * @returns {array} Objects with given datasets and corresponding assemblies and files
+ */
+const collectCloningAndMappings = (datasets, allFiles) => (
+    datasets.map((dataset) => {
+        let assemblies = dataset.assembly || [];
+        let files = [];
+        if (dataset.analyses?.length > 0) {
+            // Get the file objects for the dataset's analyses' files.
+            const analysisFilesPaths = dataset.analyses.reduce((accFiles, analysis) => accFiles.concat(analysis.files), []);
+            const cloningAndAnalysisFilePaths = _.intersection(dataset.files, analysisFilesPaths);
+            files = cloningAndAnalysisFilePaths
+                .map((filePath) => allFiles.find((file) => file['@id'] === filePath))
+                .filter((file) => file);
```

Comment:
What does this do? (`.filter((file) => file)`) I don't know if I've seen this before.

Reply:
It’s a local version of _.compact(). It just removes any falsy entries in the array, and easier than wrapping the whole RHS in _.compact().